### PR TITLE
fix: split single & bulk delete

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Separates concerns between bulk & single webhook delete. the UX was pretty strange (when clicking on the trash icon it opens a confirm but also selects in the bulk delete in the background)

### CODE REVIEW

**Comment by jhoward1994** [comment]
It looks like we need to do this at https://github.com/strapi/strapi/blob/2d79c58816fc439db3b9eb9ae4f2bdbb6848d3f5/packages/core/admin/admin/src/components/ConfirmDialog.tsx#L95-L96

(added a click handler with stopPropagation )

```
            <Button
              fullWidth
              variant="tertiary"
              onClick={(e) => {
                e.stopPropagation();
              }}
            >
```

Otherwise cancelling the single delete modal still navigates to the webhook edit page

**Review by remidej — COMMENTED** [review]
Besides what Jamie mentioned I just found one way to kind of break it:

- select a webhook
- delete it with the individual delete button
- the selection is not cleared even though the webhook doesn't exist anymore

https://github.com/strapi/strapi/assets/8087692/a5f7d1c8-d92d-4510-b32d-f407f3864805

**Comment by alexandrebodin** [comment]
@

> Besides what Jamie mentioned I just found one way to kind of break it:
> 
> * select a webhook
> * delete it with the individual delete button
> * the selection is not cleared even though the webhook doesn't exist anymore
> 
>  CleanShot.2024-06-26.at.10.49.40.mp4

Yeah I have noticed but considered it was not a pbl as it won't fail anywhere for now.

**Comment by remidej** [comment]
@alexandrebodin I agree it's not very problematic, but wouldn't it be fixed by writing `setWebhooksToDelete([])` in deleteWebhook?

**Comment by alexandrebodin** [comment]
> @alexandrebodin I agree it's not very problematic, but wouldn't it be fixed by writing `setWebhooksToDelete([])` in deleteWebhook?

I would only remove the id we just deleted from the list in case you started select multiple things and then deleted a single one. Feel free to push to the PR I have to finish sth a bit more urgent for the RC. We can always merge this another time

**Comment by remidej** [comment]
I pushed, LGTM now

File: packages/core/admin/admin/src/components/ConfirmDialog.tsx [hunk 1]
```diff
@@ -93,7 +93,13 @@ const ConfirmDialog = ({
       <Dialog.Footer>
         {startAction || (
           <Dialog.Cancel>
-            <Button fullWidth variant="tertiary">
+            <Button
```
Comment (alexandrebodin) [inline packages/core/admin/admin/src/components/ConfirmDialog.tsx:96]: is this fine to do it for all the confirms ?

**Comment by jhoward1994** [comment]
I just checked every use of <ConfirmDialog> and everything was working with this change 

[`9983fd4` (#20630)](https://github.com/strapi/strapi/pull/20630/commits/9983fd4c2a0871330913ed75696f3c610bee4ec3)

I spotted a couple of other things but can track them separately:
- Cannot close bulk CM list view publish confirm modal (it uses a different custom modal)
- Upload bulk select doesn't work

wdyt?

**Comment by alexandrebodin** [comment]
> I just checked every use of and everything was working with this change
> 
> [`9983fd4` (#20630)](https://github.com/strapi/strapi/pull/20630/commits/9983fd4c2a0871330913ed75696f3c610bee4ec3)
> 
> I spotted a couple of other things but can track them separately:
> 
> * Cannot close bulk CM list view publish confirm modal (it uses a different custom modal)
> * Upload bulk select doesn't work
> 
> wdyt?

Sounds good I'll merge this one and you can create gh issues for the others 👍

**Pull request merged**